src/main/java/com/fasterxml/jackson/databind/introspect/BasicBeanDescription.java

```diff
@@ -418,6 +418,22 @@ public AnnotatedMember findAnyGetter() throws IllegalArgumentException
         }
         return anyGetter;
     }
+    
+	public AnnotatedMember findAnySetterField() throws IllegalArgumentException {
+		AnnotatedMember anySetter = (_propCollector == null) ? null : _propCollector.getAnySetterField();
+		if (anySetter != null) {
+			/*
+			 * For now let's require a Map; in future can add support for other
+			 * types like perhaps Iterable<Map.Entry>?
+			 */
+			Class<?> type = anySetter.getRawType();
+			if (!Map.class.isAssignableFrom(type)) {
```

Comment:
Ah. So actually type is verified here. So ignore my other comments; although checks should not be done at other points if and when it is verified here already,.